### DISCUSSION TITLE
Why doesn't the component work with network URL?

### DISCUSSION THREAD

**Discussion by NetherQuartz** [whitphx/streamlit-webrtc#261] [Q&A] [answered]
Hi!
I'm trying out your component and it works like a charm being opened via local URL (localhost). But when I open it via network URL, it shows the error "Error: navigator.mediaDevices is undefined. It seems the current document is not loaded securely."
This is my code and its result:
```python
from streamlit_webrtc import ClientSettings, WebRtcMode, webrtc_streamer

WEBRTC_CLIENT_SETTINGS = ClientSettings(
    rtc_configuration={
        "iceServers": [
            {
                "urls": ["stun:stun.l.google.com:19302"]
            }
        ]
    },
    media_stream_constraints={
        "video": True,
        "audio": False,
    },
)
```
```python
webrtc_streamer(
        key="loopback",
        mode=WebRtcMode.SENDRECV,
        client_settings=WEBRTC_CLIENT_SETTINGS,
        video_processor_factory=None
    )
```
![Screenshot_20210624_034949](https://user-images.githubusercontent.com/47905455/123185729-3fc9a400-d49f-11eb-935f-903b6f91f140.png)

**Comment by whitphx** [comment, marked as answer]
Hi,
see https://dev.to/whitphx/build-a-web-based-real-time-computer-vision-app-with-streamlit-57l2#https

**Reply by NetherQuartz** [reply]
Thank you! Gonna try to connect via https.
